portal/src/main/webapp/js/src/oncoprint/studyview.js

```diff
@@ -5,10 +5,94 @@ $(document).ready(function() {
 		interpolate: /\{\{(.+?)\}\}/g
 	};
 	$('#oncoprint_controls').html(_.template($('#main-controls-template').html())());
+        
+        var clinicalElementsArray = [];
+        var GenePanelData = [];
+        var genePanelClinicals = new ClinicalColl();
+        clinicalElementsArray.push(
+        genePanelClinicals.fetch({
+        type: "POST",
+        data: {
+           cancer_study_id: cancer_study_id_selected,
+           attribute_id: "GENE_PANEL",
+           case_list: cases
+        },
+        success: function(response) {
+               GenePanelData = response.toJSON();
+           }
+        }));
+        
+        //fetch genepanel
+        var genepanelFiles = ["api/genepanel/IMPACT341","api/genepanel/IMPACT410"];
+        var genePanel = new Array();
+        var gainGenepanel = function(filename,datafetchArray)
+        {
+            datafetchArray.push(
+            $.getJSON(filename,function(result){
+                genePanel[(filename.split("/"))[2]] = result;
+            }));
+        }
 
-	var geneDataColl = new GeneDataColl();
+        for(var i=0; i<genepanelFiles.length; i++)
+        {
+            var genepanelFilename = genepanelFiles[i];
+            gainGenepanel(genepanelFilename,clinicalElementsArray);
```

Comment:
Sorry for not being clear enough, what I meant was that those lines replace your lines 38-39. And right good catch I meant ```file``` not ```name```